shopgate.php

```diff
@@ -582,22 +615,25 @@ public function hookDisplayMobileHeader()
 
 	public function hookUpdateOrderStatus($params)
 	{
-		$id_order = $params['id_order'];
-		$orderState = $params['newOrderStatus'];
+		$id_order      = $params['id_order'];
+		$orderState    = $params['newOrderStatus'];
 		$shopgateOrder = PSShopgateOrder::instanceByOrderId($id_order);
 
 		$shopgateConfig = new ShopgateConfigPresta(
-			Configuration::get('SHOPGATE_CONFIG') ?
+			is_array(unserialize(Configuration::get('SHOPGATE_CONFIG'))) ?
 				unserialize(Configuration::get('SHOPGATE_CONFIG')) :
-				array()
+				array ()
```

Comment:
Here again, please store the result in a temporary variable instead of unserializing two times